05_class_definition/02_simple_mock.rb

```diff
@@ -37,3 +37,53 @@
 # obj.imitated_method #=> true
 # obj.called_times(:imitated_method) #=> 2
 # ```
+
+module SimpleMock
+  class << self
+    def new
+      obj = Object.new
+      include_mock(obj)
+    end
+
+    def mock(obj)
+      include_mock(obj)
+    end
+
+    private
+
+    def include_mock(obj)
+      obj.singleton_class.class_eval do
+        include SimpleMock
+      end
```

Comment:
`extends`メソッドで代替可能。
[Object#extend (Ruby 3.3 リファレンスマニュアル)](https://docs.ruby-lang.org/ja/latest/method/Object/i/extend.html)

```suggestion
      obj.extends(SimpleMock)
```